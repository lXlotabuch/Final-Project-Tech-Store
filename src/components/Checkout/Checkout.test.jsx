/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react'
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { CheckoutComponent } from './Checkout';

jest.mock('./OrderSummary/OrderSummary', () => 'order-summary')
jest.mock('./FormCheckout/FormCheckout', () => 'form-checkout')

describe('CheckoutComponent', () => {
  test('render', () => {
    const {asFragment} = render(<CheckoutComponent
      isLogin
    />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('handle go to SignIn', () => {
    const onClickSignIn = jest.fn()
    const history = createMemoryHistory()

    render(
      <Router history={history}>
        <CheckoutComponent
          isLogin={false}
          onClickSignIn={onClickSignIn}
        />
      </Router>
    )
    fireEvent.click(screen.getByRole('button', {name: /sign in/i}))

    expect(history.location.pathname).toBe('/signin')
  })

  test('handle go to Cart', () => {
    const onClickCart = jest.fn()
    const history = createMemoryHistory()

    render(
      <Router history={history}>
        <CheckoutComponent
          isLogin
          onClickCart={onClickCart}
        />
      </Router>
    )
    fireEvent.click(screen.getByRole('button', {name: /back to cart/i}))

    expect(history.location.pathname).toBe('/cart')
  })
})
