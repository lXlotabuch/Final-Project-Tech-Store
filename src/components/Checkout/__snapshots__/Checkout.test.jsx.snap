// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutComponent render 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCfw sc-pFZIQ jYBaYX bvRMiD"
  >
    <div
      class="sc-bdfBwQ sc-hKgILt dkvZGM ejUGax"
    >
      <h3
        class="sc-jrAGrp UbNtR"
      >
        Checkout
      </h3>
      <div
        class="sc-gKsewC ixmeZn"
      >
        <button
          class="ant-btn ant-btn-round sc-iBPRYJ dQKGWT"
          color="borderGrey"
          type="button"
        >
          <span>
            Back to cart
          </span>
        </button>
      </div>
    </div>
    <div
      class="sc-bdfBwQ sc-dlfnbm dkvZGM gxjyut"
    >
      <order-summary />
      <div
        class="sc-iqHYGH dHHPOO"
      >
        <form-checkout />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
